examples/vision/text_image_denoiser.py

```diff
@@ -0,0 +1,280 @@
+"""
+Title: Text Image Denoiser.
+Author: [Anish B](https://twitter.com/anishhacko)
+Date created: 2021/01/18
+Last modified: 2022/01/27
+Description: Example of Text Image Denoiser coupled with Tesseract OCR engine to improve text readability.
+"""
+"""
+## Introduction
+
+This Example explains a simple **Text Image Denoiser for OCR** using U-Net based
+Architectutre. **Autoencoders** are mostly employed for **Image Restoration** problems
+but in recent times **U-Net** based architectures with **skip-connections** have gained
+popularity for several Image-to-Image tasks. We aim to solve the problem with simple
+**Pretrained U-Net as Encoder** and **Efficient Sub-Pixel CNN as Decoder**. The problem
+we focus here is enhancing the Document Images which are deteriorated through external
+degradations like blur, corrupt text blocks etc. We have followed up on few seminal
+papers which are presented below for reference. At the end of this tutorial user will
+gain clear understanding of building custom Image Pre-Processors using Deep-Learning that
+helps us to mitigate real-world OCR issues.
+
+**References:**
+- [Enhancing OCR Accuracy with Super
+Resolution](https://cdn.iiit.ac.in/cdn/cvit.iiit.ac.in/images/ConferencePapers/2018/ocr_Ankit_Lat_ICPR_2018.pdf)
+- [Improving the Perceptual Quality of Document
+Images Using Deep Neural
+Network](http://mile.ee.iisc.ac.in/publications/softCopy/DocumentAnalysis/ISNN_11page_65.pdf)
+"""
+
+"""
+## Setup
+"""
+
+# !pip install pybind11
+# !pip install fastwer
+# !pip install pytesseract
+# !sudo apt install tesseract-ocr
+
+import fastwer
+import pytesseract
+import numpy as np
+from glob import glob
+import tensorflow as tf
+from tensorflow import keras
+import matplotlib.pyplot as plt
+from tensorflow.keras import callbacks
+from tensorflow.keras.applications import resnet_v2
```

Comment:
@anish9 (IMO) It can be beautified, general packages and ml packages. For example

```python
import fastwer
import pytesseract
import numpy as np
from glob import glob
import matplotlib.pyplot as plt

import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import callbacks
from tensorflow.keras.applications import resnet_v2
```

Reply:
+1